### PR TITLE
Allow specifying domain name and github account name

### LINKED ISSUE #27: Support for repos outside of data-8
Discussed briefly with @tonyyanga. Some connector instructors expressed interest in this feature, for use with their personal repos that are outside of the data-8 organization.

### PR DESCRIPTION
**Pull request by tonyyanga**
Closes #27.

After `$ export  ALLOWED_GITHUB_ACCOUNTS=data-8:tonyyanga`, the github account specification can be tested with this example url:

`/interact?account=tonyyanga&repo=android-ad&branch=master&path=README.md`

### CODE REVIEW

**Comment by tonyyanga** [comment]
Closes #3 as well.

**Comment by tonyyanga** [comment]
@SaladRaider Just addressed what we have discussed on Trello.

**Comment by tonyyanga** [comment]
This pull request also fixes the following issues in the past deployments:

pulling from url will throw an error by default settings

verification for url domain is flawed.

File: nbinteract/pull_from_remote.py [hunk 1]
```diff
@@ -85,18 +84,18 @@ def pull_from_remote(**kwargs):
 
     # Generate repo url
     repo_url = "https://"
```
Comment (SaladRaider) [inline nbinteract/pull_from_remote.py]: I should have mentioned this earlier. Could you refactor the following code to something like `repo_url = _generate_repo_url(domain, account, repo_name)` so there isn't too much code in `pull_from_remote`?

**Review by SaladRaider — CHANGES_REQUESTED** [review]
Looks very good! I just requested some small refactoring changes.

**Comment by tonyyanga** [comment]
@SaladRaider 

Did some limited refactoring. `pull_from_remote` is still a long function, because I think most verification logic should be kept in a central place.

**Review by SaladRaider — APPROVED** [review]

**Pull request merged**